src/psrsearch.jl

```diff
@@ -38,26 +36,25 @@ julia> z_n_binned([10., 0., 0., 0., 0.], 2)
 40.0
 ```
 """
-function z_n_binned(profile::Array{T,1}, n::Integer)::T where {T<:AbstractFloat}
+function z_n_binned(profile::AbstractVector{T}, n::Integer)::T where {T<:AbstractFloat}
     total = sum(profile)
     N = length(profile)
-    phase = Array(range(0, stop = N - 1) / N) * 2 * pi
+    phase = range(0, stop = N - 1) / N * 2 * pi
 
-    if total == 0
-        return 0
+    if iszero(total)
+        return zero(T)
     end
 
-    z = 0
+    z = zero(T)
     for k in range(1, stop = n)
-        s = 0
-        for i in range(1, stop = N)
-            s += profile[i] * sin(k * phase[i])
+        s = zero(T)
+        c = zero(T)
+        for i in eachindex(profile)
```

Comment:
[`eachindex`](https://docs.julialang.org/en/v1/base/arrays/#Base.eachindex) is a convenient function that lets you forget about the indices of the iterable.